src/app/core/services/kafka.service.ts

```diff
@@ -156,73 +140,75 @@ export class KafkaService {
   }
 
   createPayloadAndSend(specs, kafkaConnInstance) {
-    let schemaVersions
-    switch (specs.name) {
-      case KAFKA_COMPLETION_LOG:
-        if (this.schemas[specs.name]) {
-          schemaVersions = this.schemas[specs.name]
-          break
-        }
-      default:
-        schemaVersions = this.util
-          .getLatestKafkaSchemaVersions(specs)
-          .catch(error => {
-            console.log(error)
-            return Promise.resolve()
-          })
-        this.schemas[specs.name] = schemaVersions
-    }
-    return Promise.all([schemaVersions]).then(data => {
-      schemaVersions = data[0]
-      const avroKey = AvroSchema.parse(
-        JSON.parse(schemaVersions[0]['schema']),
-        {
-          wrapUnions: true
-        }
-      )
-      const avroVal = AvroSchema.parse(
-        JSON.parse(schemaVersions[1]['schema']),
-        {
-          wrapUnions: true
-        }
-      )
-      const kafkaObject = specs.kafkaObject
-      const bufferKey = avroKey.clone(kafkaObject.key, { wrapUnions: true })
-      const bufferVal = avroVal.clone(kafkaObject.value, { wrapUnions: true })
-      const payload = {
-        key: bufferKey,
-        value: bufferVal
+    return this.util.getKafkaTopic(specs).then(topic => {
+      let schemaVersions
+      switch (specs.name) {
+        case KAFKA_COMPLETION_LOG:
+          if (this.schemas[specs.name]) {
+            schemaVersions = this.schemas[specs.name]
+            break
+          }
+        default:
+          schemaVersions = this.util
+            .getLatestKafkaSchemaVersions(topic)
+            .catch(error => {
+              console.log(error)
+              return Promise.resolve()
+            })
+          this.schemas[specs.name] = schemaVersions
       }
-      const schemaId = new KafkaRest.AvroSchema(
-        JSON.parse(schemaVersions[0]['schema'])
-      )
-      const schemaInfo = new KafkaRest.AvroSchema(
-        JSON.parse(schemaVersions[1]['schema'])
-      )
-      return this.sendToKafka(
-        specs,
-        schemaId,
-        schemaInfo,
-        payload,
-        kafkaConnInstance
-      ).catch(error => {
-        console.error(
-          'Could not initiate kafka connection ' + JSON.stringify(error)
+      return Promise.all([schemaVersions]).then(data => {
```

Comment:
You can just use ```Promise(schemaVersions).then (data => {
}```

Reply:
Yeah, I think Joris has already done some refactoring in his `enable-reminders` PR like here: https://github.com/RADAR-base/RADAR-Questionnaire/pull/229/files#diff-9fb9db3f68ed9b78ff47b34076f3813aR148 

So I was thinking maybe we can just use that? It should be merged after the IOS build PR is merged.